src/audio/coreaudio/SDL_coreaudio.m

```diff
@@ -1151,6 +1151,142 @@ output device (in which case we'll try again). */
+{
+    AVAudioSession* session = [AVAudioSession sharedInstance];
+    
+    if (name != NULL) {
+        *name = NULL;
+    }
+    SDL_zerop(spec);
+    spec->freq = [session sampleRate];
+    spec->channels = [session outputNumberOfChannels];
+    return 0;
+}
+#else /* MACOSX_COREAUDIO */
+static int
+COREAUDIO_GetDefaultAudioInfo(char **name, SDL_AudioSpec *spec, int iscapture)
+{
+    AudioDeviceID devid;
+    AudioBufferList *buflist;
+    OSStatus result;
+    UInt32 size;
+    CFStringRef cfstr;
+    char *devname;
+    int usable;
+    double sampleRate;
+
+    AudioObjectPropertyAddress addr = {
+        iscapture ? kAudioHardwarePropertyDefaultInputDevice
+                  : kAudioHardwarePropertyDefaultOutputDevice,
+        iscapture ? kAudioDevicePropertyScopeInput
+                  : kAudioDevicePropertyScopeOutput,
+        kAudioObjectPropertyElementMaster
+    };
+    AudioObjectPropertyAddress nameaddr = {
+        kAudioObjectPropertyName,
+        iscapture ? kAudioDevicePropertyScopeInput
+                  : kAudioDevicePropertyScopeOutput,
+        kAudioObjectPropertyElementMaster
+    };
+    AudioObjectPropertyAddress freqaddr = {
+        kAudioDevicePropertyNominalSampleRate,
+        iscapture ? kAudioDevicePropertyScopeInput
+                  : kAudioDevicePropertyScopeOutput,
+        kAudioObjectPropertyElementMaster
+    };
+    AudioObjectPropertyAddress bufaddr = {
+        kAudioDevicePropertyStreamConfiguration,
+        iscapture ? kAudioDevicePropertyScopeInput : kAudioDevicePropertyScopeOutput,
+        kAudioObjectPropertyElementMaster
+    };
+    
+    /* Get the Device ID */
+    cfstr = NULL;
+    size = sizeof (AudioDeviceID);
+    result = AudioObjectGetPropertyData(kAudioObjectSystemObject, &addr,
+                                        0, NULL, &size, &devid);
+
+    if (result != noErr) {
+        return SDL_SetError("%s: Default Device ID not found", "coreaudio");
+    }
+
+    /* Use the Device ID to get the name */
+    size = sizeof (CFStringRef);
+    result = AudioObjectGetPropertyData(devid, &nameaddr, 0, NULL, &size, &cfstr);
+
+    if (result != noErr) {
+        return SDL_SetError("%s: Default Device Name not found", "coreaudio");
+    }
+
+    CFIndex len = CFStringGetMaximumSizeForEncoding(CFStringGetLength(cfstr),
+                                                    kCFStringEncodingUTF8);
+    devname = (char *) SDL_malloc(len + 1);
+    usable = ((devname != NULL) &&
+              (CFStringGetCString(cfstr, devname, len + 1, kCFStringEncodingUTF8)));
+    CFRelease(cfstr);
+
+    if (usable) {
+        usable = 0;
+        len = strlen(devname);
+        /* Some devices have whitespace at the end...trim it. */
+        while ((len > 0) && (devname[len - 1] == ' ')) {
+            len--;
+            usable = len;
+        }
+    }
+
+    if (usable) {
+        devname[len] = '\0';
+    }
+
+    if (name != NULL) {
+        *name = devname;
+    }
```

Comment:
We could probably find some clever way to avoid allocating the devname if `name == NULL`, but for now this will patch it up:
```suggestion
    } else {
        SDL_free(devname);
    }
```